### ISSUE TITLE
JS/TS features crashing

### ISSUE THREAD

**Issue by kyletolle** [microsoft/vscode#200415]
Type: <b>Bug</b>

Not really sure what's going on but yesterday the TS support in VSCode stopped working for my project.

VS Code version: Code 1.84.2 (Universal) (1a5daa3a0231a0fbba4f14db7ec463cf99d7768e, 2023-11-09T10:52:33.687Z)
OS version: Darwin arm64 23.1.0
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Apple M3 Pro (12 x 24)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|4, 5, 5|
|Memory (System)|36.00GB (2.04GB free)|
|Process Argv|--crash-reporter-id 5138c0a8-bc8d-4e8e-83b9-195756a02068|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (40)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-nvm|abu|0.0.1
vscode-neovim|asv|1.2.0
vscode-intelephense-client|bme|1.10.1
npm-intellisense|chr|1.4.5
path-intellisense|chr|2.8.5
vscode-eslint|dba|2.4.2
svg-gallery|dev|2.1.0
EditorConfig|Edi|0.16.4
vsc-material-theme|Equ|34.3.0
vsc-material-theme-icons|equ|3.2.0
prettier-vscode|esb|10.1.0
vscode-relative-line-numbers|ext|0.0.5
vscode-open-in-github|fab|1.3.0
copilot|Git|1.141.0
copilot-chat|Git|0.10.2
vscode-pull-request-github|Git|0.76.1
search-node-modules|jas|1.3.0
ts-debug|kak|0.0.6
vscode-phpfmt|kok|1.1.35
php-namespace-resolver|Meh|1.1.9
vscode-language-babel|mgm|0.0.40
dotenv|mik|1.0.1
vscode-docker|ms-|1.28.0
python|ms-|2023.22.0
vscode-pylance|ms-|2023.11.10
remote-wsl|ms-|0.81.8
vscode-typescript-next|ms-|5.4.20231207
php-docblocker|nei|2.7.0
material-icon-theme|PKi|4.32.0
tsimporter|pmn|2.0.1
vscode-commons|red|0.0.6
vscode-xml|red|0.26.1
vscode-yaml|red|1.14.0
jenkinsfile-support|sec|0.1.0
ruby-lsp|Sho|0.5.4
move-ts|str|1.12.0
haml|vay|1.0.1
php-debug|xde|1.34.0
markdown-all-in-one|yzh|3.5.1
php-intellisense|zob|1.3.3

(1 theme extensions excluded)

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
vserr242cf:30382550
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
vscorecescf:30445987
vscod805cf:30301675
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593cf:30376535
pythonvs932:30410667
py29gd2263:30899288
vsclangdf:30486550
c4g48928:30535728
dsvsc012cf:30540253
azure-dev_surveyone:30548225
2e4cg342:30602488
f6dab269:30613381
showlangstatbar:30737416
fixshowwlkth:30771522
showindicator:30805244
pythongtdpath:30769146
i26e3531:30792625
welcomedialogc:30910334
pythonnosmt12:30797651
pythonidxpt:30866567
pythonnoceb:30805159
asynctok:30898717
dsvsc013:30795093
dsvsc014:30804076
dsvsc015:30845448
pythontestfixt:30902429
pyreplss1:30897532
pythonmypyd1:30879173
pythoncet0:30885854
2e7ec940:30885897
pythontbext0:30879054
accentitlementst:30887150
dsvsc016:30899300
dsvsc017:30899301
dsvsc018:30899302
aa_t_chat:30882232
dsvsc019:30917259

```

</details>

**Comment by VSCodeTriageBot** [comment]
Thanks for creating this issue! It looks like you may be using an old version of VS Code, the latest stable release is 1.85.0. Please try upgrading to the latest version and checking whether this issue remains.

Happy Coding!

**Comment by kyletolle** [comment]
Updating the JSC version did not work.

**Comment by kyletolle** [comment]
It is perpetually stuck trying to init.
<img width="119" alt="image" src="https://github.com/microsoft/vscode/assets/260869/5783e712-7f65-4871-8499-a67b00aec0e6">
And eventually shows it's crashed more than 5 times in 5 minutes via a notification within VSC.

**Comment by kyletolle** [comment]
When I run tsc in my project, I get an error like
```
<--- JS stacktrace --->

FATAL ERROR: Reached heap limit Allocation failed - JavaScript heap out of memory
```

So this might be an issue I have to find out with typescript more generally that's causing an issue.

**Comment by kyletolle** [comment]
This was a me-problem. I am working on an application that processes videos. Some of those files are Transport Segment (`.ts`) files. I recently synced a folder within my docker container with some of these video files to my host machine for troubleshooting. These path for these `.ts` video files were not excluded in the `tsconfig.json` for the project, so that caused the typescript build to fail, as well as prevented this VSC extension from starting up. 🤦 

After excluding that folder with the `.ts` video files, things are working again. Closing this.
